web/streetmanager.php

```diff
@@ -0,0 +1,113 @@
+            $refresh = $memcache->get($prefix . 'refresh');
+            if ($refresh) {
+                $result = $this->call('party/refresh', 'POST', [
+                    'refresh_token' => $refresh
+                ]);
+                $token = $result->id_token;
+                $memcache->set($prefix . 'token', $token, 0, 3600);
+            } else {
+                $result = $this->call('work/authenticate', 'POST', [
+                    'emailAddress' => $username,
+                    'password' => $password
+                ]);
+                $token = $result->idToken;
+                $refresh = $result->refreshToken;
+                $memcache->set($prefix . 'token', $token, 0, 3600);
+                $memcache->set($prefix . 'refresh', $refresh, 0, 86400);
+            }
+        }
+        return $token;
+    }
+
+    public function call($url, $method, $data) {
+        $data = http_build_query($data);
+        $ch = curl_init();
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+        if ($method === 'GET') {
+            curl_setopt($ch, CURLOPT_URL, $this->base . $url . '?' . $data);
+        } else {
+            curl_setopt($ch, CURLOPT_URL, $this->base . $url);
+            curl_setopt($ch, CURLOPT_POSTFIELDS, $data);
+        }
+        if ($this->token) {
+            curl_setopt($ch, CURLOPT_HTTPHEADER, ["Token: " . $this->token]);
+        }
+        $output = curl_exec($ch);
+        curl_close($ch);
+        return json_decode($output);
+    }
+}
+
+# ---
+
+$api = new \Api(API_URL, USERNAME, PASSWORD, MEMCACHE_PREFIX);
+
+$bbox = isset($_GET['bbox']) ? explode(',', $_GET['bbox']) : [];
+if (!count($bbox)) { $bbox = [405000, 286000, 408000, 287000]; }
+
+$params = [
+    'minEasting' => $bbox[0],
+    'minNorthing' => $bbox[1],
+    'maxEasting' => $bbox[2],
+    'maxNorthing' => $bbox[3],
+    'start_date' => date('Y-m-d'),
+    'end_date' => date('Y-m-d', time()+86400*13*7),
+];
+
+$data = $api->call('geojson/works', 'GET', $params);
+foreach ($data->features as $feature) {
+    $props = $feature->properties;
+    $feature->properties = [
+        'start_date' => $props->start_date,
+        'end_date' => $props->end_date,
+        'summary' => $props->work_category . ', ' . $props->traffic_management_type,
+        'description' => ucwords(strtolower($props->promoter_organisation)),
+        'promoter' => ucwords(strtolower($props->promoter_organisation)),
+    ];
+}
+
+$data2 = $api->call('geojson/activities', 'GET', $params);
+foreach ($data2->features as $feature) {
+    $props = $feature->properties;
+    $feature->properties = [
+        'start_date' => $props->start_date,
+        'end_date' => $props->end_date,
+        'summary' => $props->activity_name . ' ' . $props->activity_type,
+        'description' => $props->activity_location_description,
+    ];
+}
+$data->features = array_merge($data->features, $data2->features);
+
+$data3 = $api->call('geojson/forward-plans', 'GET', $params);
```

Comment:
At least one cobrand didn't want to display forward plans so it would be good to be able to specify in the query parameters which layers you wanted to be returned.